hoss/spatial.py

```diff
@@ -187,6 +209,71 @@ def get_projected_x_y_index_ranges(
     return x_y_index_ranges
 
 
+def get_x_y_index_ranges_from_coordinates(
+    non_spatial_variable: str,
+    varinfo: VarInfoFromDmr,
+    prefetch_coordinate_datasets: Dataset,
+    coordinates: Set[str],
+    index_ranges: IndexRanges,
+    bounding_box: BBox = None,
+    shape_file_path: str = None,
+) -> IndexRanges:
+    """This function returns a dictionary containing the minimum and maximum
+    index ranges for a pair of lat/lon coordinates, e.g.:
+
+    index_ranges = {'/x': (20, 42), '/y': (31, 53)}
+
+    This method is called when the CF standards are not followed
+    in the source granule and only coordinate datasets are provided.
+    The coordinate datasets along with the crs is used to calculate
+    the x-y projected dimension scales. The dimensions of the input,
+    non-spatial variable are checked for associated coordinates. If
+    these are present, and they have not already been added to the
+    `index_ranges` cache, the extents of the input spatial subset
+    are determined in these projected coordinates. The minimum and
+    maximum values are then derived from these projected coordinate
+    points.
+
+    """
+    crs = get_variable_crs(non_spatial_variable, varinfo)
+
+    projected_x = 'projected_x'
+    projected_y = 'projected_y'
```

Comment:
For granules with multiple grids, these names will clash for both grids.

Reply:
I think you likely want some nice naming function (likely deriving the names from the coordinate variables used), and then that function can be used in `add_index_range` to determine which entries in the cache to use.

Reply:
This is still a significant issue. If a user requests variables from both the SMAP L3 global and polar grids, this function will try to assign the x and y dimension index ranges for both to keys called `projected_x` and `projected_y` in the cache of index ranges.